### PR TITLE
Gatekeeper removal

### LINKED ISSUE #6007: Gatekeeper deployment removal on flag disable
**User Story**
Implement gatekeeper deployment removal from the user cluster when the OPA Integration flag is set to false.

### PR DESCRIPTION
**Pull request by lsviben**
**What this PR does / why we need it**:
Removes Gatekeeper from the user cluster when the user disables OPA integration. Also removes some unnecessary rbacs that were being deployed for gatekeeper

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #6007 

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: pkg/controller/seed-controller-manager/kubernetes/resources.go [hunk 1]
```diff
@@ -483,3 +475,13 @@ func (r *Reconciler) ensureStatefulSets(ctx context.Context, c *kubermaticv1.Clu
 
 	return reconciling.ReconcileStatefulSets(ctx, creators, c.Status.NamespaceName, r.Client, reconciling.OwnerRefWrapper(resources.GetClusterRef(c)))
 }
+
+func (r *Reconciler) ensureOPAIntegrationIsRemoved(ctx context.Context, data *resources.TemplateData) error {
+	for _, resource := range gatekeeper.GetResourcesToRemoveOnDelete(data.Cluster().Status.NamespaceName) {
+		if err := r.Client.Delete(ctx, resource); err != nil && !errors.IsNotFound(err) {
+			return fmt.Errorf("failed to ensure OPA integration is removed/not present: %v", err)
```
Comment (moadqassem) [inline pkg/controller/seed-controller-manager/kubernetes/resources.go:482]: Would be cool to add also the name of the resource that's being deleted in the message :)

File: pkg/controller/seed-controller-manager/kubernetes/resources.go [hunk 2]
```diff
@@ -483,3 +475,13 @@ func (r *Reconciler) ensureStatefulSets(ctx context.Context, c *kubermaticv1.Clu
 
 	return reconciling.ReconcileStatefulSets(ctx, creators, c.Status.NamespaceName, r.Client, reconciling.OwnerRefWrapper(resources.GetClusterRef(c)))
 }
+
+func (r *Reconciler) ensureOPAIntegrationIsRemoved(ctx context.Context, data *resources.TemplateData) error {
+	for _, resource := range gatekeeper.GetResourcesToRemoveOnDelete(data.Cluster().Status.NamespaceName) {
+		if err := r.Client.Delete(ctx, resource); err != nil && !errors.IsNotFound(err) {
```
Comment (moadqassem) [inline pkg/controller/seed-controller-manager/kubernetes/resources.go:481]: You might also need to check if the NotFound error is reported

File: pkg/controller/user-cluster-controller-manager/resources/reconciler.go [hunk 3]
```diff
@@ -655,3 +663,13 @@ type reconcileData struct {
 	userSSHKeys   map[string][]byte
 	cloudConfig   []byte
 }
+
+func (r *reconciler) ensureOPAIntegrationIsRemoved(ctx context.Context) error {
+	for _, resource := range gatekeeper.GetResourcesToRemoveOnDelete() {
+		if err := r.Client.Delete(ctx, resource); err != nil && !errors.IsNotFound(err) {
+			return fmt.Errorf("failed to ensure OPA integration is removed/not present: %v", err)
```
Comment (moadqassem) [inline pkg/controller/user-cluster-controller-manager/resources/reconciler.go:688]: Would be cool to add also the name of the resource that's being deleted in the message

File: pkg/controller/user-cluster-controller-manager/resources/reconciler.go [hunk 4]
```diff
@@ -655,3 +663,13 @@ type reconcileData struct {
 	userSSHKeys   map[string][]byte
 	cloudConfig   []byte
 }
+
+func (r *reconciler) ensureOPAIntegrationIsRemoved(ctx context.Context) error {
+	for _, resource := range gatekeeper.GetResourcesToRemoveOnDelete() {
+		if err := r.Client.Delete(ctx, resource); err != nil && !errors.IsNotFound(err) {
```
Comment (moadqassem) [inline pkg/controller/user-cluster-controller-manager/resources/reconciler.go:687]: You might also need to check if the NotFound error is reported

**Review by moadqassem — COMMENTED** [review]
Looks good, just minor comments

File: pkg/controller/user-cluster-controller-manager/resources/reconciler.go (hunk 4 above)
Comment (lsviben) [reply pkg/controller/user-cluster-controller-manager/resources/reconciler.go:687]: Its checking for NotFound already, or did you mean something else?

Comment (moadqassem) [reply pkg/controller/user-cluster-controller-manager/resources/reconciler.go:687]: Yes you are right, sorry I overlooked that

**Pull request merged**